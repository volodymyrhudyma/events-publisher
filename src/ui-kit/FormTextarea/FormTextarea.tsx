import React, { ChangeEvent, PureComponent } from 'react';
import { IFormControlProps } from 'src/types';
import FormControl from 'src/ui-kit/FormControl';
import Textarea from 'src/ui-kit/Textarea';

interface IFormTextareaProps extends IFormControlProps {
	placeholder: string;
	value: string;
	onChange: (e: ChangeEvent<HTMLTextAreaElement>) => void;
}

interface IFormTextareaState {
	length: number;
}

class FormTextarea extends PureComponent<IFormTextareaProps, IFormTextareaState> {
	public static defaultProps = {
		required: false
	};
	constructor(props: IFormTextareaProps) {
		super(props);
		this.state = {
			length: 0
		};
	}
	public onLengthChange = (length: number) => {
		this.setState({
			...this.state,
			length
		});
	};
	public onChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
		const { onChange, maxLength } = this.props;
		if (maxLength) {
			const { length } = e.target.value;
			if (length <= maxLength) {
				this.onLengthChange(length);
				onChange(e);
			}
		} else {
			onChange(e);
		}
	};
	public render() {
		const { placeholder, value, ...rest } = this.props;
		const { length } = this.state;
		return (
			<FormControl {...rest} currentLength={length}>
				<Textarea value={value} onChange={this.onChange} placeholder={placeholder} />
			</FormControl>
		);
	}
}

export default FormTextarea;
