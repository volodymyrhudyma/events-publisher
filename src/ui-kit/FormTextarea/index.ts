import Component from './FormTextarea';

export default Component;
