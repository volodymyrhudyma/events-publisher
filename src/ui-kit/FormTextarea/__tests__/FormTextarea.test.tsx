import React from 'react';
import { shallow } from 'enzyme';
import FormTextarea from '../FormTextarea';

describe('FormTextarea component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormTextarea
        label="Label"
        placeholder="Placeholder"
        value=""
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
