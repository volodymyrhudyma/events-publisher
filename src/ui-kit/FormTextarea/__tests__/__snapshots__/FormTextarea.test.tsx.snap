// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextarea component should render correctly 1`] = `
<FormControl
  currentLength={0}
  label="Label"
  onChange={[MockFunction]}
  required={false}
>
  <Textarea
    onChange={[Function]}
    placeholder="Placeholder"
    rows={10}
    value=""
  />
</FormControl>
`;
